File: Cmdlets/Models/LookupCounterItem.vb
Imports System.Collections.Generic

Public Class LookupCounterItem(Of TKey, TValue)

	Public Sub New(Key As TKey, Count As Integer, Values As IEnumerable(Of TValue))
		Me.Key = Key
		Me.Count = Count
		Me.Values = Values
	End Sub

	Public ReadOnly Property Key() As TKey

	Public ReadOnly Property Count() As Integer

	Public ReadOnly Property Values() As IEnumerable(Of TValue)
End Class

File: Common/Models/LookupCounterItem.cs
using System.Collections.Generic;

namespace TIKSN.Cmdlets.Common.Models
{
    public class LookupCounterItem<TKey, TValue>
    {
        public LookupCounterItem(TKey Key, int Count, IEnumerable<TValue> Values)
        {
            this.Key = Key;
            this.Count = Count;
            this.Values = Values;
        }

        public int Count { get; }
        public TKey Key { get; }
        public IEnumerable<TValue> Values { get; }
    }
}